src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,64 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static String splitDelimiter;
+    private static String target;
+    public static int splitAndSum(String text) throws Exception {
+        splitDelimiter = "[,:]";
+        target = text;
+
+        if(isBlank(target))
+            return 0;
+
+        changeDelimiterAndTargetIfAvailable(text);
+
+        if(isSingleInteger(target))
+            return Integer.parseInt(target);
+
+        List<String> numbers = Arrays.asList(target.split(splitDelimiter));
+
+        if(hasInappropriateValue(numbers))
+            throw new RuntimeException();
+
+        return numbers.stream().mapToInt(Integer::parseInt).sum();
+    }
+
+    private static boolean hasInappropriateValue(List<String> numbers) {
+        return numbers.stream().filter(n -> Integer.parseInt(n) < 0).findAny().isPresent()
+                || numbers.stream().filter(n -> !isNumeric(n)).findAny().isPresent();
+    }
+
+    private static void changeDelimiterAndTargetIfAvailable(String text) {
+        Matcher matcher = Pattern.compile("//(.)\n(.*)").matcher(text);
+
+        if(matcher.find()){
+            splitDelimiter = matcher.group(1);
+            target = matcher.group(2);
+        }
+    }
+
+    private static boolean isSingleInteger(String text) {
+        if(text.split(splitDelimiter).length != 1)
+            return false;
+        if(!isNumeric(text))
+            throw new RuntimeException();
+
+        return true;
+    }
```

Comment:
```suggestion
    private static boolean isSingleInteger(String text) {
    if (!isNumeric(text))
        throw new RuntimeException();

    return text.split(splitDelimiter).length == 1;
}
```
요렇게도 해볼 수 있을거 같아요! 😄

Reply:
@parkeeseul
와... 그렇네요 ㅋㅋㅋ 감사합니다

Reply:
@parkeeseul
위와 같이 수정했는데 test case에서 에러가 나서.. 봤더니
Single이 아닌 Input의 경우(예를들어, "1,2,3")
isNumeric() 메서드에서 파싱오류가 발생해서 RuntimeException()을 던져버리네요.. ㅠㅠ

isNumeric()의 파라미터를 text.split(splitDelimiter)로 받아서 isNumeric()도 같이 수정해야 할 것 같습니다!